packages/postcss-merge-longhand/src/lib/decl/columns.js

```diff
@@ -60,10 +60,10 @@ function explode(rule) {
 
     values.forEach((value, i) => {
       let prop = properties[1];
-
+      const dimension = unit(value);
       if (value.toLowerCase() === auto) {
         prop = properties[i];
-      } else if (unit(value).unit) {
+      } else if (dimension && dimension.unit !== '') {
```

Comment:
I don't know whether the original code skipped this branch on purpose if `unit === ''` or if it was just an oversight. I have written the check like this to preserve the same output.